src/index.js

```diff
@@ -1,41 +1,134 @@
 const express = require('express');
 const cors = require('cors');
 
-// const { v4: uuidv4 } = require('uuid');
+const { v4: uuidv4 } = require('uuid');
 
 const app = express();
 
 app.use(cors());
 app.use(express.json());
 
-// const users = [];
+const users = [];
 
 function checksExistsUserAccount(request, response, next) {
-  // Complete aqui
+  const { username } = request.headers;
+
+
+  const usuario = users.find(user => user.username === username);
+  
+
+  if (!usuario) {
+    return response.status(404).json({Erro: "usuario não existe"})
+  }
+
+  request.usuario = usuario;
+
+  return next()
+
 }
 
 app.post('/users', (request, response) => {
-  // Complete aqui
+  const { name, username } = request.body
```

Comment:
It would be nice to have a more defensive approach here. if the request does not have the required fields, your controller's action will break. For this you can use [try-catch](https://javascript.info/try-catch).